wdl-engine/src/eval.rs

```diff
@@ -149,3 +201,155 @@ impl<'a> ScopeRef<'a> {
+            command: execution.command().into(),
+            stdout,
+            stderr,
+            outputs: Ok(Default::default()),
+        })
+    }
+
+    /// Gets the status code of the evaluated task.
+    pub fn status_code(&self) -> i32 {
+        self.status_code
+    }
+
+    /// Gets the working directory of the evaluated task.
+    pub fn work_dir(&self) -> &Path {
+        &self.work_dir
+    }
+
+    /// Gets the temp directory of the evaluated task.
+    pub fn temp_dir(&self) -> &Path {
+        &self.temp_dir
+    }
+
+    /// Gets the command file of the evaluated task.
+    pub fn command(&self) -> &Path {
+        &self.command
+    }
+
+    /// Gets the stdout value of the evaluated task.
+    pub fn stdout(&self) -> &Value {
+        &self.stdout
+    }
+
+    /// Gets the stderr value of the evaluated task.
+    pub fn stderr(&self) -> &Value {
+        &self.stderr
+    }
+
+    /// Gets the outputs of the evaluated task.
+    ///
+    /// This is `Ok` when the task executes successfully and all of the task's
+    /// outputs evaluated without error.
+    ///
+    /// Otherwise, this contains the error that occurred while attempting to
+    /// evaluate the task's outputs.
+    pub fn outputs(&self) -> &EvaluationResult<Outputs> {
+        &self.outputs
+    }
+
+    /// Converts the evaluated task into an evaluation result.
+    ///
+    /// Returns `Ok(_)` if the task outputs were evaluated.
+    ///
+    /// Returns `Err(_)` if the task outputs could not be evaluated.
+    pub fn into_result(self) -> EvaluationResult<Outputs> {
+        self.outputs
+    }
+
+    /// Handles the exit of a task execution.
+    ///
+    /// Returns an error if the task failed.
+    fn handle_exit(&self, requirements: &HashMap<String, Value>) -> anyhow::Result<()> {
+        let mut error = true;
+        if let Some(return_codes) = requirements
+            .get("return_codes")
+            .or_else(|| requirements.get("returnCodes"))
+        {
+            match return_codes {
+                Value::Primitive(PrimitiveValue::String(s)) if s.as_ref() == "*" => {
+                    error = false;
+                }
+                Value::Primitive(PrimitiveValue::String(s)) => {
+                    bail!("invalid return code value `{s}`; only `*` is accepted");
+                }
+                Value::Primitive(PrimitiveValue::Integer(ok)) => {
+                    if self.status_code == i32::try_from(*ok).unwrap_or_default() {
+                        error = false;
+                    }
+                }
+                Value::Compound(CompoundValue::Array(codes)) => {
+                    error = !codes.as_slice().iter().any(|v| {
+                        v.as_integer()
+                            .map(|i| i32::try_from(i).unwrap_or_default() == self.status_code)
+                            .unwrap_or(false)
+                    });
+                }
+                _ => unreachable!("unexpected return codes value"),
```

Comment:
I'm assuming `Value` implements `Display`, in which case, would it be wise to print out the exact text that isn't as expected here?

Reply:
We can print it out in the panic, but this should be `unreachable!` due to requiring an error-free static analysis for the task evaluation to even occur.

Reply:
Oh I see. Okay I think I glossed over that this was unreachable. No need to print it out then.